src/csharp/Grpc.Core.Api/AsyncStreamReaderExtensions.cs.cs

```diff
@@ -0,0 +1,47 @@
+#region Copyright notice and license
+
+// Copyright 2015 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System.Collections.Generic;
+using System.Threading;
+
+namespace Grpc.Core
```

Comment:
I like that the extension method is immediately available to someone writing a gRPC service via the Grpc.Core namespace. If the namespace was Grpc.Core.Util then they would have to know to add it.